ale_linters/javascript/jshint.vim

```diff
@@ -0,0 +1,50 @@
+if exists('g:loaded_ale_linters_javascript_jshint')
+    finish
+endif
+
+let g:loaded_ale_linters_javascript_jshint = 1
+
+function! ale_linters#javascript#jshint#Handle(buffer, lines)
+    " Matches patterns line the following:
+    "
+    " stdin:57:9: Missing name in function declaration.
+    " stdin:60:5: Attempting to override 'test2' which is a constant.
+    " stdin:57:10: 'test' is defined but never used.
+    " stdin:57:1: 'function' is defined but never used.
+    let pattern = '^.\+:\(\d\+\):\(\d\+\): \(.\+\)'
+    let output = []
+
+    for line in a:lines
+        let l:match = matchlist(line, pattern)
+
+        if len(l:match) == 0
+            continue
+        endif
+
+        let text = l:match[3]
+        let marker_parts = l:match[4]
+
+        if len(marker_parts) == 2
+            let text = text . ' (' . marker_parts[1] . ')'
+        endif
+
+        " vcol is Needed to indicate that the column is a character.
+        call add(output, {
+        \   'bufnr': a:buffer,
+        \   'lnum': l:match[1] + 0,
+        \   'vcol': 0,
+        \   'col': l:match[2] + 0,
+        \   'text': text,
+        \   'type': 'E',
+        \   'nr': -1,
+        \})
+    endfor
+
+    return output
+endfunction
+
+call ALEAddLinter('javascript', {
+\   'executable': 'jshint',
+\   'command': 'jshint --reporter unix --config ' . g:ale_jshint_config_loc . ' -',
```

Comment:
You should use a default for the config location or make the config option in some way. I now offer a `command_callback` option which lets you generate the command string which will be run in a job. The function takes a buffer number as its only argument. You can look at the shell linter I just added around an hour ago for an example of its use.

Thank you for adding more tools! You contribution is much appreciated. I need to add an option for selecting which linters you want, which is part of why I now require a `'name'` option in the dictionary here for identifying which linter it is for which filetype. I'll add that once my RSI clears up. I've been having some issues recently.